packages/graphile-build-pg/src/plugins/PgTypesPlugin.js

```diff
@@ -997,31 +1000,58 @@ end`;
           type.type === "d" &&
           type.domainBaseTypeId
         ) {
-          const baseType = getGqlTypeByTypeIdAndModifier(
-            type.domainBaseTypeId,
-            typeModifier
-          );
-          const baseInputType = getGqlInputTypeByTypeIdAndModifier(
-            type.domainBaseTypeId,
-            typeModifier
-          );
-          // Hack stolen from: https://github.com/graphile/postgraphile/blob/ade728ed8f8e3ecdc5fdad7d770c67aa573578eb/src/graphql/schema/type/aliasGqlType.ts#L16
-          gqlTypeByTypeIdAndModifier[type.id][typeModifierKey] = Object.assign(
-            Object.create(baseType),
-            {
-              name: inflection.domainType(type),
-              description: type.description,
-            }
-          );
-          if (baseInputType && baseInputType !== baseType) {
+          // might be used as an enum alias: #1500
+          if (
+            type.name.endsWith("_enum_domain") ||
+            (type.tags && type.tags.enum)
+          ) {
+            const underlyingEnumTypeName = type.name.endsWith("_enum_domain")
+              ? type.name.replace("_enum_domain", "")
+              : type.tags.enum;
+
+            const baseTypeId = `FAKE_ENUM_${type.namespaceName}_${underlyingEnumTypeName}`;
```

Comment:
This feels very brittle; can we instead use getGqlTypeByPgTypeAndModifier (or similar) passing the type ID for the referenced enum table? This allows me to change the enum identifier without keeping particular lines of code in sync.

Reply:
Based on my investigation, the ID for the referenced enum table wasn't actually available in the introspection results. Looks like it's substituted here: https://github.com/graphile/graphile-engine/blob/v4/packages/graphile-build-pg/src/plugins/PgIntrospectionPlugin.js#L580 and here https://github.com/graphile/graphile-engine/blob/v4/packages/graphile-build-pg/src/plugins/PgIntrospectionPlugin.js#L557

Reply:
You might be right; I’ll check later in the week 👍

Reply:
Yeah you're right; let's stick with using the enum fake name. With that in mind, please could you extract the generation of this value into a function that can be called in both places; something like:

```js
const fakeEnumIdentifier = (klass, constraint, list = false) => {
  const constraintIdent = constraint.type === "p" ? "" : `_${constraint.name}`;
  const baseEnumIdentifier = `FAKE_ENUM_${klass.namespaceName}_${klass.name}${constraintIdent}`;
  if (list) {
    return `${baseEnumIdentifier}_list`;
  } else {
    return baseEnumIdentifier;
  }
}
```

This'll make it less brittle. Please put this helper on Build so external plugins can use it too. I'm kinda tempted to put something random into it to stop people using the true value in their plugins, but I think that ship has already sailed.

Reply:
Done in https://github.com/graphile/graphile-engine/pull/748/commits/5f1f5693d33fc034123a19691a6bd5318458222c

Reply:
Excellent thanks!